github/teams.go

```diff
@@ -203,6 +203,52 @@ func (s *TeamsService) EditTeam(ctx context.Context, id int64, team NewTeam) (*T
 	return t, resp, nil
 }
 
+// RemoveParentTeam represents a new team with no parent
+type RemoveParentTeam struct {
+	Name         string   `json:"name"` // Name of the team. (Required.)
+	Description  *string  `json:"description,omitempty"`
+	Maintainers  []string `json:"maintainers,omitempty"`
+	RepoNames    []string `json:"repo_names,omitempty"`
+	ParentTeamID *int64   `json:"parent_team_id,omitempty"`
```

Comment:
I'm sorry, but I don't understand.

How is this struct different from the `NewTeam` struct on line 138 of teams.go?

Have you tested this out on GitHub and does it work?

Additionally, it seems like a lot of code duplication (especially as it is currently written).

It seems like an argument could be added (maybe `removeParent bool`) to the existing `EditTeam` call, and then the resulting JSON body could simply add in the field: `"parent_team_id": null`, right? 

Wouldn't that be much easier to maintain than copy/pasting all the surrounding code and tweaking one line?

Reply:
If we set the parent_team_id to `nil`, due to the `omitempty` it will be [removed](https://play.golang.org/p/Z8WhKO_Q_LK). Changing the buffer in parsing defies the logic for that method. Instead of converting changing the buffer, we can parse it to a different struct and pass on that to parse json in `EditTeam` function.